assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,752 @@
+        // Construct updated imageSource builder image object based on it's kind for the only imageSource
+        $scope.updatedBuildConfig.spec.source.images[0].from = $scope.constructImageObject($scope.imageSourceOptions);
+      }
+
+      // Construct updated builder image object based on it's kind
+      if ($scope.builderOptions.pickedType === "None") {
+        delete $filter('buildStrategy')($scope.updatedBuildConfig).from
+      } else {
+        $filter('buildStrategy')($scope.updatedBuildConfig).from = $scope.constructImageObject($scope.builderOptions);
+      }
+
+      // Construct updated output image object based on it's kind. Only Image Stream Tag, Docker Image and None can 
+      // be specified for the output image. Not Image Stream Image since they are immutable.
+      if ($scope.outputOptions.pickedType === "None") {
+        // If user will change the output reference to 'None' shall the potential PushSecret be deleted as well?
+        // This case won't delete them.
+        delete($scope.updatedBuildConfig.spec.output.to)
+      } else {
+        $scope.updatedBuildConfig.spec.output.to = $scope.constructImageObject($scope.outputOptions);
+      }
+
+      // Update envVars
+      $filter('buildStrategy')($scope.updatedBuildConfig).env = $scope.updateEnvVars($scope.envVars);
+
+      // Update triggers
+      $scope.updatedBuildConfig.spec.triggers = $scope.updateTriggers();
+
+      DataService.update("buildconfigs", $scope.updatedBuildConfig.metadata.name, $scope.updatedBuildConfig, {
+        namespace: $scope.updatedBuildConfig.metadata.namespace
+      }).then(
+        function() {
+          AlertMessageService.addAlert({
+            name: $scope.updatedBuildConfig.metadata.name,
+            data: {
+              type: "success",
+              message: "Build Config " + $scope.updatedBuildConfig.metadata.name + " was sucessfully updated."
+            }
+          });
+          $location.path(Navigate.resourceURL($scope.updatedBuildConfig, "BuildConfig", $scope.updatedBuildConfig.metadata.namespace));
+        },
+        function(result) {
+          $scope.disableInputs = false;
+
+          $scope.alerts["save"] = {
+            type: "error",
+            message: "An error occurred updating the build " + $scope.updatedBuildConfig.metadata.name + "Build Config",
+            details: $filter('getErrorDetails')(result)
+          }
+        }
+      );
+    };
+
+    $scope.isNamespaceAvailable = function(namespace) {
+      if ($scope.availableProjects.contains(namespace)) {
+        return true;
+      }
+      return false;
+    };
+
+    $scope.inspectNamespace = function(imageStreams, pickedImageStream) {
+      if (imageStreams.length === 0) {
+        return "empty";
+      } 
+      if (imageStreams.length !== 0 && !imageStreams.contains(pickedImageStream)) {
+        return "noMatch";
+      }
+      return "";
```

Comment:
Feels like these should be two different methods that return booleans rather than one that returns string.